src/main/java/com/github/calebwhiting/runelite/plugins/actionprogress/detect/GuardianOfTheRift.java

```diff
@@ -0,0 +1,144 @@
+	private static final int DIALOG_WIDGET_GROUP = 229;
+	private static final int DIALOG_WIDGET_MESSAGE = 1;
+
+	private int currentElementalRewardPoints = -1;
+	private int currentCatalyticRewardPoints = -1;
+
+	private boolean triggeringEssenceCrafting;
+	private boolean triggeringRewardSearch;
+
+	@Inject private Client client;
+
+	@Inject private InventoryManager inventoryManager;
+
+	@Inject protected InterruptManager intterupManager;
+
+	private boolean checkInMainRegion(){
+		int[] currentMapRegions = client.getMapRegions();
+		return Arrays.stream(currentMapRegions).anyMatch(x -> x == MINIGAME_MAIN_REGION);
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage chatMessage)
+	{
+		if(!checkInMainRegion()) return;
+		if(chatMessage.getType() != ChatMessageType.SPAM && chatMessage.getType() != ChatMessageType.GAMEMESSAGE) return;
+
+		String msg = chatMessage.getMessage();
+		Matcher rewardPointMatcher = REWARD_POINT_PATTERN.matcher(msg);
+		
+		if(msg.contains("You found some loot:")){
+			currentElementalRewardPoints--;
+			currentCatalyticRewardPoints--;
+		}
+		else if(msg.contains("The Great Guardian was defeated!")){
+			intterupManager.interrupt("Guardian of the Rift game concluded");
+		}
+		else if(rewardPointMatcher.find()) {
+			intterupManager.interrupt("Guardian of the Rift game concluded");
+			currentElementalRewardPoints = Integer.parseInt(rewardPointMatcher.group(1).replaceAll(",", ""));
+			currentCatalyticRewardPoints = Integer.parseInt(rewardPointMatcher.group(2).replaceAll(",", ""));
+		}
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked evt)
+	{
+		if(!checkInMainRegion()) return;
+
+		if (evt.getId() == REWARD_GUARDIAN_ID && evt.getMenuOption().toLowerCase().equals("search")){
+			triggeringRewardSearch = true;
+		}
+
+		if(evt.getId() == WORKBENCH_ID && evt.getMenuOption().toLowerCase().equals("work-at")){			
+			triggeringEssenceCrafting = true;
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick evt)
+	{
+		if(!checkInMainRegion()) return;
+
+		Widget dialog = client.getWidget(DIALOG_WIDGET_GROUP, DIALOG_WIDGET_MESSAGE);
+		if (dialog != null){			
+		String dialogText = dialog.getText();
+		final Matcher checkMatcher = CHECK_POINT_PATTERN.matcher(dialogText);
+		if (checkMatcher.find(0))
+		{
+			currentCatalyticRewardPoints = Integer.parseInt(checkMatcher.group(1));
+			currentElementalRewardPoints = Integer.parseInt(checkMatcher.group(2));
+			}
+		}
+		
+		Player local = this.client.getLocalPlayer();
+		if (local == null || local.getAnimation() == -1) return;
+
+		log.debug (local.getAnimation() + " local animation");
+		
+		if (triggeringEssenceCrafting && local.getAnimation() == CRAFTING_ANIMATION_ID){
+			triggeringEssenceCrafting = false;
+
+			ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+			
+			int numberOfGuardianEssencesToMake = inventory == null ? 0 : Math.min(inventory.count(GUARDIAN_FRAGMENTS_ITEM_ID), this.inventoryManager.getFreeSpaces());
```

Comment:
As mentioned on discord, I recommend moving this to a class variable that is initiated [here](https://github.com/guillaume009/runelite-plugin-action-progress/pull/31/files#diff-4ef2fb4cc62da02e79973c8930e6edf9076bd2ba63b04336b829b8e6dce669fdR97) to prevent an inventory miscount issue.

Reply:
Done :)